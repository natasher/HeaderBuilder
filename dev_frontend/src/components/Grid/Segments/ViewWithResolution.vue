<template>
  <span class = "title">
    {{ getCurrentView | capitalize }}<span class="res">{{ getCurrentBreakpoint }}</span>
  </span>
</template>

<script>
import { mapState   } from 'vuex'
import { capitalize } from '../../../filters/index'

export default {

  name: 'ViewWithResolution',

  computed: {
    ...mapState( 'general', {
      getCurrentView      : state => state.currentView,
    }),

    ...mapState( 'ui', {
      getCurrentBreakpoint: state => state.currentBreakpoint,
    }),
  },

  filters: {
    capitalize,
  },

}
</script>
