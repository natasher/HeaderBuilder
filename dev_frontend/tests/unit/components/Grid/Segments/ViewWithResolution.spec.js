import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vuex               from       'vuex'
import merge              from       'lodash.merge'
import ViewWithResolution from       '../../../../../src/components/Grid/Segments/ViewWithResolution.vue'

const localVue = createLocalVue()
localVue.use( Vuex )

const brk  = '= 666'
const view = 'testView'

function createStore(overrides) {
  const defaultStoreOptions = {
    modules: {
      general: {
        namespaced: true,
        state: {
          currentView: view
        }
      },
      ui: {
        namespaced: true,
        state: {
          currentBreakpoint: brk
        },
      },
    },
  }

  return new Vuex.Store( merge(defaultStoreOptions, overrides))
}

function createWrapper(overrides) {
  const defaultMountingOptions = {
    localVue,
    store: createStore(),
  }

  return shallowMount( ViewWithResolution, merge(defaultMountingOptions, overrides))
}

describe('ViewWithResolution.vue', () => {

  test('should match snapshot', () => {
    const wrapper = createWrapper()

    expect( wrapper.element ).toMatchSnapshot()
  })

  test('should render capitalized view name', () => {
    const wrapper = createWrapper()

    expect( wrapper.text() ).toContain('TestView')
  })

  test('should render breakpoint', () => {
    const wrapper = createWrapper()

    expect( wrapper.text() ).toContain( brk )
  })

})