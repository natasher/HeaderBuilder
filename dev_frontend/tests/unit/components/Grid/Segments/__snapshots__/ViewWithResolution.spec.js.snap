// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewWithResolution.vue should match snapshot 1`] = `
<span
  class="title"
>
  
  TestView
  <span
    class="res"
  >
    = 666
  </span>
</span>
`;
